src/Builder.js

```diff
@@ -34,6 +67,9 @@ Builder.prototype = {
     var terms = [];
     for (var idx = 0; idx < arguments.length; idx++) {
       var arg = arguments[idx];
+      if (!(arg instanceof pexprs.PExpr)) {
```

Comment:
It would be nice to avoid having to do this in every one of the constructor functions. Is there a reason that you can't do this from the bottom up, by calling `toRecipe` on all the arguments before invoking the constructor function?